pkg/cloud/azure/actuators/machine/actuator.go

```diff
@@ -178,22 +215,44 @@ func (a *Actuator) Update(ctx context.Context, cluster *clusterv1.Cluster, machi
 		return a.handleMachineError(machine, apierrors.UpdateMachine("failed to create machine %q scope: %v", machine.Name, err), updateEventAction)
 	}
 
-	err = a.reconcilerBuilder(scope).Update(context.Background())
-	if err != nil {
-		// We still want to persist on failure to update MachineStatus
-		if err := scope.Persist(); err != nil {
-			klog.Errorf("Error storing machine info: %v", err)
+	vm, err := a.reconcilerBuilder(scope).Update(context.Background())
+	if vm != nil {
+		modMachine, err := a.setMachineCloudProviderSpecifics(machine, *vm)
+		if err != nil {
+			klog.Errorf("%s: error updating machine cloud provider specifics: %v", machine.Name, err)
+		} else {
+			machine = modMachine
+		}
+
+		modMachine, err = a.updateStatus(ctx, machine, a.reconcilerBuilder(scope), *vm)
+		if err != nil {
+			klog.Infof("%s: failed to update status: %v", machine.Name, err)
+			return &controllerError.RequeueAfterError{RequeueAfter: requeueAfterSeconds * time.Second}
+		}
+		machine = modMachine
+
+		modMachine, err = a.updateProviderID(machine, *vm, scope.SubscriptionID, scope.ClusterConfig.ResourceGroup)
+		if err != nil {
+			klog.Infof("%s: failed to set provider ID: %v", machine.Name, err)
+		} else {
+			machine = modMachine
 		}
+
+		modMachine, updateConditionError := a.updateMachineProviderConditions(machine, providerconfig.MachineCreated, machineCreationSucceedReason, machineCreationSucceedMessage)
```

Comment:
This does not belong under Update. No need to update any condition in Update op.

Reply:
Wrong, it does. Create does not have to finish and set the MachineCreated condition. In which case the Update op has to do it.